crates/sui-graphql-client/src/lib.rs

```diff
@@ -0,0 +1,568 @@
+pub struct SuiClient {
+    /// The URL of the GraphQL server.
+    rpc: String,
+    /// GraphQL supports multiple versions of the service. By default, the client uses the stable
+    /// version of the service.
+    rpc_version: Option<String>,
+    inner: ReqwestHttpClient,
+}
+
+impl Default for SuiClient {
+    /// Create a new [`SuiClient`] with testnet as the default GraphQL server.
+    fn default() -> Self {
+        Self {
+            rpc: TESTNET_HOST.to_string(),
+            rpc_version: None,
+            inner: ReqwestHttpClient::new(),
+        }
+    }
+}
+
+impl SuiClient {
+    // ===========================================================================
+    // Client Misc API
+    // ===========================================================================
+
+    /// Set the server address for the GraphQL GraphQL client. It should be a valid URL with a host and
+    /// optionally a port number.
+    pub fn set_rpc_server(&mut self, server: &str) -> Result<(), anyhow::Error> {
+        reqwest::Url::parse(server)?;
+        self.rpc = server.to_string();
+        Ok(())
+    }
+
+    /// Set the version for the GraphQL GraphQL client. The default set version is stable.
+    ///
+    /// Use the `service_config` API to get the `available_versions` field to know which versions
+    /// are supported by this service.
+    pub fn set_version(&mut self, version: Option<&str>) {
+        self.rpc_version = version.map(|v| v.to_string());
+    }
+
+    /// Return the URL for the GraphQL GraphQL client after checking if the version is set.
+    fn url(&self) -> String {
+        if let Some(version) = &self.rpc_version {
+            format!("{}/{}", self.rpc, version)
+        } else {
+            self.rpc.to_string()
+        }
+    }
+
+    /// Set the GraphQL to localhost and port 9125.
+    pub fn set_localhost(&mut self) -> &Self {
+        self.rpc = DEFAULT_LOCAL_HOST.to_string();
+        self
+    }
+
+    /// Set GraphQL client to devnet.
+    pub fn set_devnet(&mut self) -> &Self {
+        self.rpc = DEVNET_HOST.to_string();
+        self.set_version(None);
+        self
+    }
+
+    /// Set GraphQL client to testnet.
+    pub fn set_testnet(&mut self) -> &Self {
+        self.rpc = TESTNET_HOST.to_string();
+        self
+    }
+
+    /// Set GraphQL client to mainnet.
+    pub fn set_mainnet(&mut self) -> &Self {
+        self.rpc = MAINNET_HOST.to_string();
+        self
+    }
+
+    /// Run a query on the GraphQL server and return the response.
+    /// This method returns an [`cynic::GraphQlResponse`]  over the query type `T`, and it is
+    /// intended to be used with custom queries.
+    pub async fn run_query<
+        T: serde::de::DeserializeOwned + Send,
+        V: serde::Serialize + Send + std::marker::Sync,
+    >(
+        &self,
+        operation: &Operation<T, V>,
+    ) -> Result<GraphQlResponse<T>> {
+        self.inner.post(&self.url(), operation).await
+    }
```

Comment:
```suggestion
    pub async fn run_query<T, V>(
        &self,
        operation: &Operation<T, V>,
    ) -> Result<GraphQlResponse<T>>
    where
        T: serde::de::DeserializeOwned,
        V: serde::Serialize,
    {
        self.inner.post(&self.url(), operation).await
    }
```

lets use a where clause, its a bit easier to read. And are the send/Sync reqs coming from Operation? I don't think they're strictly necessary.

Reply:
I do not know to be honest, compiler is complaining so I added them as suggested by the compiler.